index/scorch/introducer.go

```diff
@@ -241,3 +252,44 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 	// notify merger we incorporated this
 	close(nextMerge.notify)
 }
+
+func (s *Scorch) revertToSnapshot(revertTo *snapshotReversion) error {
```

Comment:
interestingly, this implementation is very general... which is nice.  As long as the caller can somehow get their hands on an IndexSnapshot, this method actually just writes (a copy of) that IndexSnapshot as the new root.

So, there's perhaps an opportunity to get a better name for this method / functionality...

It's almost like WriteSnapshot(*IndexSnapshot) error or AcceptSnapshot().  Naming is hard.

Reply:
ForceSnapshot()?  PushSnapshot()?